src/unlock/omni_lock.rs

```diff
@@ -133,6 +172,91 @@ bitflags! {
     }
 }
 
+impl Serialize for SmtProofEntryVec {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_newtype_struct("SmtProofEntryVec", &self.as_bytes())
+    }
+}
+
+impl<'de> Deserialize<'de> for SmtProofEntryVec {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let bytes = Bytes::deserialize(deserializer)?;
+        SmtProofEntryVec::from_slice(bytes.as_ref()).map_err(|e| {
+            serde::de::Error::invalid_value(
+                Unexpected::Bytes(bytes.as_ref()),
+                &format!(
+                    "can not convert the value to SmtProofEntryVec:　{}",
+                    e.to_string()
+                )
+                .as_str(),
+            )
+        })
+    }
+}
+
+impl hash::Hash for SmtProofEntryVec {
+    fn hash<H>(&self, state: &mut H)
+    where
+        H: hash::Hasher,
+    {
+        self.as_slice().hash(state);
+    }
+}
+
+// impl Eq
+
+impl PartialEq for SmtProofEntryVec {
+    fn eq(&self, other: &SmtProofEntryVec) -> bool {
+        self.as_slice() == other.as_slice()
+    }
+}
+impl Eq for SmtProofEntryVec {}
+
+/// The administrator mode configuration.
+#[derive(Clone, Serialize, Deserialize, Debug, Hash, Eq, PartialEq, Default)]
+pub struct AdminConfig {
+    /// The rc cell's type script hash, the type script should be a type id script.
+    rc_type_id: H256,
+    /// The smt proofs
+    proofs: SmtProofEntryVec,
+    /// The alternative auth content to the args part.
+    auth: Option<Identity>,
+}
+
+impl AdminConfig {
+    pub fn set_rc_type_id(&mut self, rc_type_id: H256) {
+        self.rc_type_id = rc_type_id;
+    }
+    pub fn set_proofs(&mut self, proofs: SmtProofEntryVec) {
+        self.proofs = proofs;
+    }
+    pub fn rc_type_id(&self) -> &H256 {
+        &self.rc_type_id
+    }
+    pub fn proofs(&self) -> &SmtProofEntryVec {
+        &self.proofs
+    }
+
+    /// set the additional auth, it will be used to sign the transaction.
+    pub fn set_auth(&mut self, auth: Identity) {
+        self.auth = Some(auth);
+    }
+
+    pub fn new(root: H256, proofs: SmtProofEntryVec) -> AdminConfig {
+        AdminConfig {
+            rc_type_id: root,
+            proofs,
+            ..Default::default()
```

Comment:
```suggestion
            auth: None,
```
Since `Default::default()` will construct a new struct of `AdminConfig`.